packages/wallets/src/solana/services/approvals-service.ts

```diff
@@ -41,4 +50,21 @@ export class SolanaApprovalsService {
         }
         return transaction;
     }
+
+    private retrieveValidSignature(signedTxn: VersionedTransaction) {
+        let signatureObj = null;
+        for (let i = 0; i < signedTxn.signatures.length; i++) {
```

Comment:
you can also find the signature that corresponds to that signer like this: https://github.com/Paella-Labs/crossbit-main/pull/18179/files#diff-783c98152b8acbc0153ec1f0049113d0881919bbd1815cf3b0de98439f4e71ebR201